### PR TITLE
refactor: remove `ObjectUplaoder` and turn stream uploading

### PR DESCRIPTION
**Pull request by trim21**
ref https://github.com/minio/minio-js/issues/1045

previously, if user want to upload a stream, for example: `s3.putObject('...', '...', fs.createReadStream('...'))`, minio-js will use `maxObjectSize` as object size and 500 mb as chunk size.

after this patch, when `putObject` is used with a stream created by `fs.createReadStream`, minio-js will try to read file size as object size, and use proper chunk size.

### CODE REVIEW

**Comment by trim21** [comment]
@prakashsvmx

**Review by aldy505 — COMMENTED** [review]
LGTM overall, but here are some nitpicks.

File: src/internal/helper.ts [hunk 1]
```diff
@@ -586,3 +597,30 @@ export function parseXml(xml: string): any {
...
+
+/**
+ * get content size of object content to upload
+ */
+export async function getContentLength(s: stream.Readable | Buffer | string): Promise<number | null> {
+  // use length property of string | Buffer
+  const length = (s as unknown as Record<string, unknown>).length as number | undefined
+  if (isNumber(length)) {
+    return length
+  }
```
[the remark this answers is not in the document]
Comment (trim21) [reply src/internal/helper.ts:609]: this suggestion would fail `tsc`

**Review by aldy505 — COMMENTED** [review]
LGTM

**Review by prakashsvmx — APPROVED** [review]
Tested. Changes look good to me. :+1:

**Comment by trim21** [comment]
@kaankabalak

**Review by kaankabalak — APPROVED** [review]

**Pull request merged**